### ISSUE TITLE
After upgrading to 2023.4.1, integration no longer able to start. 

### ISSUE THREAD

**Issue by calum-mcfarlane** [oziee/ha-solcast-solar#117]
### The problem

Integration had been set up a while back and was working fine. After I restarted HA after upgrading to 2023.4.1, the integration failed to start. 

I restored the backup to go back to 2023.4.0 but this did not resolve the problem.

### What version of Solcast Integration has the issue?

3.0.45

### What version of Home Assistant are you using?

HAOS in a PVE VM

### Anything in the logs that might be useful for us?

```txt
This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/solcast_solar/coordinator.py:79
Integration: Solcast PV Forecast (documentation, issues)
First occurred: 09:39:40 (1 occurrences)
Last logged: 09:39:40

Error setting up entry Solcast Solar for solcast_solar
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/solcast_solar/__init__.py", line 40, in async_setup_entry
    await coordinator.setup()
  File "/config/custom_components/solcast_solar/coordinator.py", line 79, in setup
    await get_instance(self._hass).async_add_executor_job(self.gethistory)
asyncio.exceptions.CancelledError
```

### Additional information

If I restart HA the system seems to block on starting this integration for several minutes, before giving up and flagging the "failed to set up" error on the integrations dashboard. I've enabled debug logging but it doesn't seem to change what's logged about this issue.I've also not been able to download the diagnostics info,  the option doesn't show up in the menu.

**Comment by oziee** [comment]
I am not seeing any of these errors in the latest version in HA latest version
I also don't see how you can't download the diagnostic info as its built into HA

<img width="526" alt="Screenshot 2023-04-10 at 6 19 37 pm" src="https://user-images.githubusercontent.com/1471841/230861197-87218d38-9ffb-42d6-ae17-0a4fb58858b7.png">

make sure its the latest version.. click the three dots to get the diagnostic file

sounds like your HA installation is having problems if things are blocking and not loading correctly

**Comment by kikorr** [comment]
Same here:

2023-04-11 17:21:44.780 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Solcast Solar for solcast_solar
File "/config/custom_components/solcast_solar/__init__.py", line 38, in async_setup_entry
File "/config/custom_components/solcast_solar/coordinator.py", line 78, in setup
2023-04-11 17:24:33.907 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - trying to connect to - https://api.solcast.com.au/rooftop_sites?format=json&api_key=REDACTED
2023-04-11 17:24:34.556 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - sites_data code http_session returned data type is <class 'dict'>
2023-04-11 17:24:34.557 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - sites_data code http_session returned status 200
2023-04-11 17:24:34.557 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - sites_data returned data: {'sites': [{'name': 'Casa', 'resource_id': 'db7e-8e84-f62a-1da5', 'capacity': 10, 'capacity_dc': 9.9, 'longitude': -3.XXXXX, 'latitude': 40.XXXXX, 'azimuth': 180, 'tilt': 16, 'install_date': '2022-11-08T23:00:00.0000000Z', 'loss_factor': 0.9, 'tags': []}], 'page_count': 1, 'current_page': 1, 'total_records': 1}
2023-04-11 17:24:34.568 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - load_saved_data file exists.. file type is <class 'dict'>
2023-04-11 17:26:12.917 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Solcast Solar for solcast_solar
File "/config/custom_components/solcast_solar/__init__.py", line 38, in async_setup_entry
File "/config/custom_components/solcast_solar/coordinator.py", line 78, in setup
2023-04-11 17:29:56.621 DEBUG (DbWorker_1) [custom_components.solcast_solar.coordinator] SOLCAST - gethistory got 0 items

**Comment by oziee** [comment]
@kikorr 
Not sure yours is the same at all. The error log you have pasted here does not line up with any code lines in the current version of this integration 

Make sure you are running the latest version of the integration

**Comment by kikorr** [comment]
Maria

**Comment by oziee** [comment]
ok so a few questions..

-have you changed the sensor name? the integration is hard coded looking for the sensor named `sensor.solcast_forecast_this_hour` to get past data from
-if using Maria DB there is a DB conversion that runs in the background by HA to update their changes to the DB system.. this can take hours and could be causing a problem with getting past data.
- @kikorr your not even running the latest version of the integration as your log code shows your using old code and no the latest. Make sure you update the integration to see if that helps or not
- @kikorr you have hijacked the issue here but you issue is different, your log shows that it is getting the past data as it logs out  `SOLCAST - gethistory got 0 items` so the code ran.. again update to the latest version of the integration

I have just added some extra catch code to help catch any errors in getting past data to allow the integration to continue functioning if it happens v3.0.46

**Comment by kikorr** [comment]
If I should start a new Issue let me know.

v.3.0.46 installed through HACS.
![image](https://user-images.githubusercontent.com/708794/231452686-1e3bedd9-625c-4403-b4ff-1371e4ee9ac3.png)


Sensor.solcast_forecast_this_hour is unavailable
![image](https://user-images.githubusercontent.com/708794/231452036-8c1e4d04-5540-43cf-918e-9ca5eb844a01.png)

![image](https://user-images.githubusercontent.com/708794/231452224-ba20403a-a567-43df-96e0-362fe4dcb07e.png)
![image](https://user-images.githubusercontent.com/708794/231452262-74890bae-5d0a-4f9f-9e70-46cf521d59b0.png)

![image](https://user-images.githubusercontent.com/708794/231452426-0dba1bb9-558f-45f6-b66c-852d8c4b4c61.png)

**Comment by oziee** [comment]
what does the full HA log show around happening to get a better idea when its occurring and what the integration is actually logging as it goes along

seems in the past most ppl with bootstraps setup times outs occur because of corrupt db files
you could try this.. **MAKE SURE YOU HAVE A FULL BACKUP FIRST :) **

most HA updates make changes the db system.. and sometimes since they can take a log time ppl start restarting the ha server when its still updating and can cause problems

if you have a full back, then delete the db files for HA and restart HA and see if the error goes away.. if it goes away its a corrupt DB file.. if its still there then restore the backup (cause then its not a corrupt db file)

**Comment by kikorr** [comment]
I completelly deletde the solcast integration before and now if I try to reinstall it shows this error so I cant do the test, no?

![image](https://user-images.githubusercontent.com/708794/231471852-010a91b0-8cde-4d24-8743-f2463e8bc4cb.png)


The Logs shows only this:



2023-04-12 15:25:46.232 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Solcast Solar for solcast_solar
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/solcast_solar/__init__.py", line 40, in async_setup_entry
    await coordinator.setup()
  File "/config/custom_components/solcast_solar/coordinator.py", line 79, in setup
    await get_instance(self._hass).async_add_executor_job(self.gethistory)
asyncio.exceptions.CancelledError

![image](https://user-images.githubusercontent.com/708794/231472295-90de6ed4-4e25-4e98-87ac-58ae76e13863.png)

**Comment by calum-mcfarlane** [comment]
Apologies for delay in replying, I was able to remove the Solcast integration, update HA to latest and then re-install the integration and now it all seems to be working again.

I'm guessing something to do with the update broke something and the re-install fixed it, since it was working before and it's working again now.

**Comment by oziee** [comment]
@calum-mcfarlane great 😃 

@kikorr not sure if it's the Maria db failing for you. Deleting the integration and reinstalling will not help if the data it reads from us corrupt.
For normal HA backing up and deleting the .db files in the /config folder and restarting clears it so you can test, but I'm not sure where the Maria db file is to rename to test

**Comment by calum-mcfarlane** [comment]
@oziee This morning I upgraded to HA 2023.4.3 and the Solcast integration didn't restart along with everything else. I was able to delete and re-install the integration and it's working again now. However it will get a bit tedious if I need to do that for every upgrade!

In the logs I see this from when HA was restarting after the upgrade:
```
This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/solcast_solar/coordinator.py:79
Integration: Solcast PV Forecast (documentation, issues)
First occurred: 08:08:55 (1 occurrences)
Last logged: 08:08:55

Error setting up entry Solcast Solar for solcast_solar
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/solcast_solar/__init__.py", line 40, in async_setup_entry
    await coordinator.setup()
  File "/config/custom_components/solcast_solar/coordinator.py", line 79, in setup
    await get_instance(self._hass).async_add_executor_job(self.gethistory)
asyncio.exceptions.CancelledError
```
Then when removing Solcast this is what was logged
```
This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/solcast_solar/__init__.py:130
Integration: Solcast PV Forecast (documentation, issues)
First occurred: 08:21:38 (1 occurrences)
Last logged: 08:21:38

Error calling entry remove callback Solcast Solar for solcast_solar
Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/__init__.py", line 124, in async_remove_entry
    coordinator: SolcastUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
KeyError: 'solcast_solar'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 579, in async_remove
    await component.async_remove_entry(hass, self)
  File "/config/custom_components/solcast_solar/__init__.py", line 130, in async_remove_entry
    _LOGGER.error("SOLCAST - async_remove_entry: %s - Err Msg: %s",traceback.format_exc(), err.message)
AttributeError: 'KeyError' object has no attribute 'message'

```
None of this tells me a great deal, I don't know why it would break when upgrading a minor version but be completely fine when removed and re-installed. I've done plenty of version upgrades before with Solcast installed no problem, it feels like something in 2023.4.1 might have changed but I have no way to know.

**Comment by oziee** [comment]
Uninstalling the integration and reinstalling - the code for the integration doesn't change. So it's not the integration. Nothing is changing with it, only HA is changing.

I have been updating HA updates and on my HA I don't have any problems lately.

HA don't make it easy when they keep changing method names for no great reason. They have also been doing a lots of changing on the db back end and if I just make this integration a forecast and ignore all previous data now of the code problem in this issue would happen as it is for you failing to get past data from HA history.

**Comment by kikorr** [comment]
Right now it is working. First of all, I would like to say that I was not saving any solcast statistics in BBDD, I had it excluded from the recorder.

We went from home assistant v 2023.4.3 and solcast unavailable to install.
![image](https://user-images.githubusercontent.com/708794/231713388-89cfb408-7002-4d01-aee1-9975afe70727.png)

to here:
![image](https://user-images.githubusercontent.com/708794/231714449-353d71ba-bd83-4c89-86f5-346d8723d7cc.png)

I could install now the integration.

and finally here:
![image](https://user-images.githubusercontent.com/708794/231713548-4a9a1872-766f-4913-b46d-56c9e3a32ca1.png)
![image](https://user-images.githubusercontent.com/708794/231713562-4ecddd84-9040-48f3-9aa1-4054b48861a2.png)
![image](https://user-images.githubusercontent.com/708794/231713606-34a94b2c-17c5-4c71-b627-2ff53d4e50e4.png)

So its seems to me that the Home Assistant Image has some kind of issue.

BTW now its all working as expected.

**Comment by angryninja48** [comment]
I have been struggling with this integration not working now for a week. I've successfully used this integration now for nearly a year without any issues however upgrading to 2023.4 lead to something breaking.
I have tried uninstalling and reinstalling, downgrading and upgrading home assistant but nothing seemed to fix this issue. I am running home assistant in a kubernetes cluster and today I disabled ipv6 on all nodes which lead to all my broken integrations now working including solcast.

TLDR; If its not working, disable ipv6.

**Comment by oziee** [comment]
IPv6 on or off has no effect on this integration. There is no code in it that requires it.

The fact turning it off things started to work for you.. great.. but again no effect

**Comment by angryninja48** [comment]
Doesn't have to be your code however I will point out that the API you target resolves to an IPv6 (as well as an IPv4) address however.

``` 
;; QUESTION SECTION:
;api.solcast.com.au.		IN	AAAA

;; ANSWER SECTION:
api.solcast.com.au.	266	IN	CNAME	d1uiw5fqppi7il.cloudfront.net.
d1uiw5fqppi7il.cloudfront.net. 60 IN	AAAA	2600:9000:2083:bc00:c:5e84:e340:93a1
d1uiw5fqppi7il.cloudfront.net. 60 IN	AAAA	2600:9000:2083:8000:c:5e84:e340:93a1
d1uiw5fqppi7il.cloudfront.net. 60 IN	AAAA	2600:9000:2083:2a00:c:5e84:e340:93a1
d1uiw5fqppi7il.cloudfront.net. 60 IN	AAAA	2600:9000:2083:7000:c:5e84:e340:93a1
d1uiw5fqppi7il.cloudfront.net. 60 IN	AAAA	2600:9000:2083:800:c:5e84:e340:93a1
d1uiw5fqppi7il.cloudfront.net. 60 IN	AAAA	2600:9000:2083:5200:c:5e84:e340:93a1
d1uiw5fqppi7il.cloudfront.net. 60 IN	AAAA	2600:9000:2083:d800:c:5e84:e340:93a1
d1uiw5fqppi7il.cloudfront.net. 60 IN	AAAA	2600:9000:2083:a200:c:5e84:e340:93a1
```

**Comment by oziee** [comment]
again nothing to do with the integration and its function what so ever. as with integrations it connects to the HA system to use network connections
